docs/providers/aws/guide/plugins.md

```diff
@@ -413,6 +413,102 @@ module.exports = MyPlugin;
 
 Command names need to be unique. If we load two commands and both want to specify the same command (e.g. we have an integrated command `deploy` and an external command also wants to use `deploy`) the Serverless CLI will print an error and exit. If you want to have your own `deploy` command you need to name it something different like `myCompanyDeploy` so they don't clash with existing plugins.
 
+### Extending validation schema
+
+If your plugin adds support for additional params in `serverless.yml` file, you should also add validation rules to the Framework's schema. Otherwise, the Framework may place validation errors to command output about your params.
+
+The Framework uses JSON-schema validation backed by [AJV](https://github.com/epoberezkin/ajv). You can extend [initial schema](/lib/configSchema/index.js) inside your plugin constuctor by modifying `serverless.configSchemaHandler.schema` property. Since the schema is a plain object dynamically loaded inside your plugin constuctor, you can modify it however you want. However in most cases you would want to use `defineCustomProperty` or `defineCustomEvent` helpers.
```

Comment:
I would probably recommend just using pre-prepared utils (and we should introduce one to extend schema on `provider`)

Reply:
Sure I'll adjust it.

I am planning to remove phrases like `you can modify it however you want` and include docs for `defineProvider` method